src/main.ts

```diff
@@ -0,0 +1,251 @@
+import { getDefaultAnnotations, getDefaultLabels } from './utils';
+import { parseCreateReleaseResponse } from './output-parser';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { version: appVersion } = require('../package.json');
+
+/**
+ * DeployCloudRunOutputs are the common GitHub action outputs created by this action
+ */
+export interface CreateCloudDeployReleaseOutputs {
+  name?: string | null | undefined; // Match clouddeploy_v1.Schema$Release.name
+  link?: string | null | undefined; // Link to console based on the above
+}
+
+/**
+ * Executes the main action. It includes the main business logic and is the
+ * primary entry point. It is documented inline.
+ */
+export async function run(): Promise<void> {
+  const restoreEnv = stubEnv({
+    CLOUDSDK_METRICS_ENVIRONMENT: 'github-actions-create-cloud-deploy-release',
+    CLOUDSDK_METRICS_ENVIRONMENT_VERSION: appVersion,
+  });
+
+  // Warn if pinned to HEAD
+  if (isPinnedToHead()) {
+    logWarning(pinnedToHeadWarning('v1'));
+  }
+  try {
+    // Core inputs (required)
+    const name = getInput('name');
+    const deliveryPipeline = getInput('delivery_pipeline');
+    const region = getInput('region');
+    const source = getInput('source');
+    const buildArtifacts = getInput('build_artifacts');
+    const images = parseKVString(getInput('images'));
+    const disableInitialRollout = getBooleanInput('disable_initial_rollout');
+    const sourceStagingDir = getInput('gcs_source_staging_dir');
+    const skaffoldFile = getInput('skaffold_file');
+    const annotations = parseKVString(getInput('annotations'));
+    const labels = parseKVString(getInput('labels'));
+    const description = getInput('description');
+    const flags = getInput('flags');
+    const gcloudComponent = presence(getInput('gcloud_component'));
+    const gcloudVersion = await computeGcloudVersion(getInput('gcloud_version'));
+
+    let cmd;
+
+    // Throw errors if inputs aren't valid
+    if (!name) {
+      throw new Error('No release name set.');
+    }
+    if (!deliveryPipeline) {
+      throw new Error('No delivery pipeline set.');
+    }
+    if (!region) {
+      throw new Error('No region set.');
+    }
+    if (!buildArtifacts && !images) {
+      throw new Error('One of `build_artifacts` and `images` inputs must be supplied.');
+    }
+    if (buildArtifacts && images && Object.keys(images).length > 0) {
+      throw new Error('Both `build_artifacts` and `images` inputs set - please select only one.');
+    }
+
+    // Validate gcloud component input
+    if (gcloudComponent && gcloudComponent !== 'alpha' && gcloudComponent !== 'beta') {
+      throw new Error(`invalid input received for gcloud_component: ${gcloudComponent}`);
+    }
+
+    // Build base command from required inputs
+    cmd = [
+      'deploy',
+      'releases',
+      'create',
+      name,
+      '--delivery-pipeline',
+      deliveryPipeline,
+      '--region',
+      region,
```

Comment:
There's an interesting discussion to have here. By passing in `--region`, we're overriding anything the user configured previously.

Reply:
For example, if the user has previously set something like `gcloud config set deploy/region europe-central1`? I agree there are various options for what should take precedence -- as I am now setting a default region in the `action.yml` file, I imagine this should override any previous configuration?

Reply:
I'm wondering if the default value for "region" in the action.yml should be `""`. Then, in this code, we can explicitly check for the empty string and print out a debug/warning message that says "no region set, will use gcloud default. If this fails, set a region in your action.yml".